content-services/content-sources-backend
# Fixes 3490: update distributions for create/update templates

## Summary
This creates/updates/deletes distributions in pulp when a template is created or updated (and repositories are added/removed).

See the JIRA card for the breakdown of the distribution paths. In summary, distributions are created under a path associated to each template. And the paths for red hat repos and custom repos are different.

## Testing steps
1. These testing steps are just a starting point, there's a lot of cases to test here.
2. Sync 2 repositories.
3. Create a template and include the first repository.
4. After the template is created, grab the distribution_href from the "templates_repository_configurations" table. Query pulp for that href. Make sure it is created and that the path matches the requirements of the JIRA card.
5. Patch the template, including the first and second repositories in the request. Check that the old repository's distribution persists and that a new one is created for the second repository.
6. Patch the template again, removing the second repository. The distribution for that repository should be deleted.
7. If you have the means to host your own repositories, you'll also want to test changing the date of the template. I'm not sure how to test this otherwise.

## Checklist
